deploy/lib/parca-agent/parca-agent.libsonnet

```diff
@@ -240,6 +240,14 @@ function(params) {
           name: 'run',
           mountPath: '/run',
         },
+        {
+          name: 'boot',
+          mountPath: '/boot',
+        },
+        {
+          name: 'proc',
+          mountPath: '/proc',
```

Comment:
It might not be a good idea to override the container's `/proc`, we may need to use something like `/host/proc`

Reply:
And I did a test on my local machine, `/proc/config.gz` is available in the containers and the options look good (even without `hostPID: true`)

Reply:
Which configuration exactly have you tested?

Reply:
I checked in a K3d cluster backed by Podman in rootless mode and in Minikube:

```console
$ kubectl run --attach --rm --image=docker.io/busybox --restart=Never "busybox-${USER}" -- \
    gunzip --stdout /proc/config.gz \
    | grep -E '^(CONFIG_(BPF(|_SYSCALL|_JIT(|_ALWAYS_ON)|_EVENTS)|HAVE_EBPF_JIT|ARCH_WANT_DEFAULT_BPF_JIT))='
CONFIG_BPF=y
CONFIG_HAVE_EBPF_JIT=y
CONFIG_ARCH_WANT_DEFAULT_BPF_JIT=y
CONFIG_BPF_SYSCALL=y
CONFIG_BPF_JIT=y
CONFIG_BPF_EVENTS=y
```

The Scaleway cluster does not seem to have the file.

Reply:
I also found a new setting that would be more appropriate, `procMount: Unmasked` in the container `securityContext`:
https://snyk.io/blog/10-kubernetes-security-context-settings-you-should-understand/#procmount

(I have not found any good official documentation about it yet though)

Reply:
For `procMount`? No, it does not seem necessary at the moment, and its alpha state makes it inconvenient

Found its design proposal, linking it here for easy reference: https://github.com/kubernetes/design-proposals-archive/blob/main/auth/proc-mount-type.md